### ISSUE TITLE
JSON parsing incorrectly in Release* modes

### ISSUE THREAD

**Issue by okx-code** [ziglang/zig#12292]
### Zig Version

0.10.0-dev.3369+fdaf9c40d

This bug is not present in version 0.9.1.

### Steps to Reproduce

```zig
const std = @import("std");

const StrangeStruct = struct {
    name: ?[]u8 = null, // it works if you remove this field
    num: ?i8 = null,
    nodes: ?[]StrangeStruct = null, // OR if you remove this one
};

pub fn main() anyerror!void {
    var messageBuf: []const u8 = "{\"num\":1}";

    var tokens = std.json.TokenStream.init(messageBuf);
    const json = try std.json.parse(StrangeStruct, &tokens, .{});

    std.log.err("Printing", .{});
    if (json.num) |num| {
        // should print the number, but doesn't in ReleaseSafe, ReleaseSmall, and ReleaseFast
        std.log.err("num {d}", .{num});
    }
}
```
Save to bug.zig and compile with `zig build-exe -O ReleaseSafe bug.zig` and `zig build-exe -O Debug bug.zig`

### Expected Behavior

I expected the `num` field to be printed in both Debug and ReleaseFast/Safe/Small modes.

### Actual Behavior

The `num` field is only printed in Debug mode.

**Comment by okx-code** [comment]
Hey @Vexu could this be marked as a regression? This code works as expected in 0.9.1

**Comment by star-tek-mb** [comment]
Maybe its not only json related. If you change your i8 type to i64 - your code works.

**Comment by r00ster91** [comment]
I think Vexu's stage1 label was correct actually because I can't reproduce this in stage2 so we can just ignore this issue and move on with replacing stage1 with stage2.

**Comment by andrewrk** [comment]
I can't reproduce this with any combination of stage1, stage2, or build mode.

**Comment by okx-code** [comment]
Looks like it was fixed somewhere between 0.10.0 3369 and 0.10.0 4640, thanks!
